tsdb/agent/db_test.go

```diff
@@ -380,9 +378,6 @@ func TestWALReplay(t *testing.T) {
 
 	s, err := Open(logger, reg, remoteStorage, promAgentDir, opts)
 	require.NoError(t, err)
-	defer func() {
-		require.NoError(t, s.Close())
-	}()
```

Comment:
Windows is not happy with this. But this `s` is getting closed. What is not getting closed in the one at line 400 `s, err = Open(...)` (after this change) which was being caught by this defer. We can bring back this defer but move it after line 401.